### ISSUE TITLE
In `ttype`, how does `List` differ from `Tuple`?

### ISSUE THREAD

**Issue by rebcabin** [rebcabin/masr#2]
I assume `List` is homogeneous and `Tuple` is heterogeneous. Otherwise, they're both ordered collections, start-index 0, with duplicate entries allowed?

```
ttype
    = Integer(int kind, dimension* dims)
...
    | Set(ttype type)
    | List(ttype type)
    | Tuple(ttype* type)
    | Struct(symbol derived_type, dimension* dims)
...
```

**Comment by certik** [comment]
> I assume `List` is homogeneous and `Tuple` is heterogeneous, as suggested (if not implied ?!?) by the arguments. Otherwise, they're both ordered collections, start-index 0, with duplicate entries allowed?

Yes. Tuple has compile time length and types for each element. List has a compile time type, the same for each element, and the length is runtime (you can append / remove from it).

The documentation for all ASR nodes is here: https://github.com/lfortran/lfortran/tree/c648a8d824242b676512a038bf2257f3b28dad3b/doc/src/asr/asr_nodes and please do not close this issue until I document this there.

**Comment by rebcabin** [comment]
roger. Pinned the link, btw. ty :)
